modules/tensor_mechanics/src/materials/FiniteStrainHyperElasticViscoPlastic.C

```diff
@@ -503,35 +504,39 @@ FiniteStrainHyperElasticViscoPlastic::updateFlowRate()
 void
 FiniteStrainHyperElasticViscoPlastic::computeQpJacobian()
 {
-  RankTwoTensor pk2fet, fepk2;
-  RankFourTensor dcedfe, dfedf, tan_mod;
-
+  _dce_dfe.zero();
   for (unsigned int i = 0; i < LIBMESH_DIM; ++i)
     for (unsigned int j = 0; j < LIBMESH_DIM; ++j)
       for (unsigned int k = 0; k < LIBMESH_DIM; ++k)
       {
-        dcedfe(i,j,k,i) += _fe(k,j);
-        dcedfe(i,j,k,j) += _fe(k,i);
+        _dce_dfe(i,j,k,i) += _fe(k,j);
+        _dce_dfe(i,j,k,j) += _fe(k,i);
       }
 
-  tan_mod = _fe.mixedProductIkJl(_fe) * _dpk2_dce * dcedfe;
-  pk2fet = _pk2[_qp] * _fe.transpose();
-  fepk2 = _fe * _pk2[_qp];
+  _tan_mod = _fe.mixedProductIkJl(_fe) * _dpk2_dce * _dce_dfe;
+  _pk2_fet = _pk2[_qp] * _fe.transpose();
+  _fe_pk2 = _fe * _pk2[_qp];
 
   for (unsigned int i = 0; i < LIBMESH_DIM; ++i)
     for (unsigned int j = 0; j < LIBMESH_DIM; ++j)
       for (unsigned int l = 0; l < LIBMESH_DIM; ++l)
       {
-        tan_mod(i,j,i,l) += pk2fet(l,j);
-        tan_mod(i,j,j,l) += fepk2(i,l);
+        _tan_mod(i,j,i,l) += _pk2_fet(l,j);
+        _tan_mod(i,j,j,l) += _fe_pk2(i,l);
       }
 
   for (unsigned int i = 0; i < LIBMESH_DIM; ++i)
     for (unsigned int j = 0; j < LIBMESH_DIM; ++j)
       for (unsigned int l = 0; l < LIBMESH_DIM; ++l)
-        dfedf(i,j,i,l) =  _fp_tmp_inv(l,j);
+        _dfe_df(i,j,i,l) =  _fp_tmp_inv(l,j);
+
+  for (unsigned int i = 0; i < LIBMESH_DIM; ++i)
+    for (unsigned int j = 0; j < LIBMESH_DIM; ++j)
+      for (unsigned int k = 0; k < LIBMESH_DIM; ++k)
+        for (unsigned int l = 0; l < LIBMESH_DIM; ++l)
+          _df_dstretch_inc(i, j, k, l) = _rotation_increment[_qp](i, k) * _deformation_gradient_old[_qp](l, j);
```

Comment:
Is this the same as `innerProductTranspose` or possibly some other function on `RankFourTensor`?